### PR TITLE
Update PHPStan version

### PR DESCRIPTION
**Pull request by comxd**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Update PHPStan version.
| Fixed ticket? | N/A

### CODE REVIEW

File: src/content/1.7/modules/testing/ci-cd.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
       # Run a container for PHPStan, having access to the module content and PrestaShop sources.
       # This tool is outside the composer.json because of the compatibility with PHP 5.6
       - name : Run PHPStan
-        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
+        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon .
```
Comment (PululuK) [inline src/content/1.7/modules/testing/ci-cd.md:85]:
```suggestion
        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon ./
```

File: src/content/1.7/modules/testing/ci-cd.md [hunk 2]
```diff
@@ -175,7 +175,7 @@ phpstan-php-7-2:
     php composer.phar install --dev
     php composer.phar global require phpstan/phpstan-shim:0.11.19
     ln -s /builds/ps-addons/$CI_PROJECT_NAME /var/www/html/modules/$CI_PROJECT_NAME
-    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
+    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon .
```
Comment (PululuK) [inline src/content/1.7/modules/testing/ci-cd.md:178]:
```suggestion
    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon ./
```

**Review by PululuK — COMMENTED** [review]
Nice catch @comxd ! Some suggestions.

**Review by PululuK — APPROVED** [review]

File: src/content/1.7/modules/testing/ci-cd.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ jobs:
       # Run a container for PHPStan, having access to the module content and PrestaShop sources.
       # This tool is outside the composer.json because of the compatibility with PHP 5.6
       - name : Run PHPStan
-        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
+        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon ./
```
Comment (Progi1984) [inline src/content/1.7/modules/testing/ci-cd.md:85]:
```suggestion
        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.12 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
```

File: src/content/1.7/modules/testing/ci-cd.md [hunk 4]
```diff
@@ -175,7 +175,7 @@ phpstan-php-7-2:
     php composer.phar install --dev
     php composer.phar global require phpstan/phpstan-shim:0.11.19
```
Comment (Progi1984) [inline src/content/1.7/modules/testing/ci-cd.md:176]:
```suggestion
    php composer.phar global require phpstan/phpstan-shim:0.12
```

File: src/content/1.7/modules/testing/ci-cd.md [hunk 5]
```diff
@@ -175,7 +175,7 @@ phpstan-php-7-2:
...
     ln -s /builds/ps-addons/$CI_PROJECT_NAME /var/www/html/modules/$CI_PROJECT_NAME
-    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
+    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon ./
```
Comment (Progi1984) [inline src/content/1.7/modules/testing/ci-cd.md:178]:
```suggestion
    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Run without  the path argument in welcome module : https://github.com/PrestaShop/welcome/blob/dev/.github/workflows/php.yml

File: src/content/1.7/modules/testing/ci-cd.md [hunk 6]
```diff
@@ -82,7 +82,7 @@ jobs:
       # Run a container for PHPStan, having access to the module content and PrestaShop sources.
       # This tool is outside the composer.json because of the compatibility with PHP 5.6
       - name : Run PHPStan
-        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
+        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.12 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline src/content/1.7/modules/testing/ci-cd.md:85]:
```suggestion
        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.12 analyse --configuration=/web/module/tests/phpstan/phpstan.neon .
```

File: src/content/1.7/modules/testing/ci-cd.md [hunk 7]
```diff
@@ -173,7 +173,7 @@ phpstan-php-7-2:
   stage: tests
   script: |
     php composer.phar install --dev
-    php composer.phar global require phpstan/phpstan-shim:0.11.19
+    php composer.phar global require phpstan/phpstan-shim:0.12
     ln -s /builds/ps-addons/$CI_PROJECT_NAME /var/www/html/modules/$CI_PROJECT_NAME
     php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline src/content/1.7/modules/testing/ci-cd.md:178]:
```suggestion
    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon .
```

**Review by PierreRambaud — APPROVED** [review]

File: src/content/1.7/modules/testing/ci-cd.md (hunk 1 above)
Comment (PierreRambaud) [reply src/content/1.7/modules/testing/ci-cd.md]: There is no reason to add the / since `.` means "current directory"

**Review by atomiix — APPROVED** [review]

File: src/content/1.7/modules/testing/ci-cd.md (hunk 6 above)
Comment (Progi1984) [reply src/content/1.7/modules/testing/ci-cd.md]: But the dot is not mandatory : 
> Run without the path argument in welcome module : https://github.com/PrestaShop/welcome/blob/dev/.github/workflows/php.yml

**Review by matthieu-rolland — APPROVED** [review]

File: src/content/1.7/modules/testing/ci-cd.md [hunk 8]
```diff
@@ -82,7 +82,7 @@ jobs:
       # Run a container for PHPStan, having access to the module content and PrestaShop sources.
       # This tool is outside the composer.json because of the compatibility with PHP 5.6
       - name : Run PHPStan
-        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.11.19 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
+        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.12 analyse --configuration=/web/module/tests/phpstan/phpstan.neon .
```
Comment (Progi1984) [inline src/content/1.7/modules/testing/ci-cd.md:85]:
```suggestion
        run: docker run --rm --volumes-from temp-ps -v $PWD:/web/module -e _PS_ROOT_DIR_=/var/www/html --workdir=/web/module phpstan/phpstan:0.12 analyse --configuration=/web/module/tests/phpstan/phpstan.neon
```

No need with 0.12 version

File: src/content/1.7/modules/testing/ci-cd.md [hunk 9]
```diff
@@ -173,9 +173,9 @@ phpstan-php-7-2:
   stage: tests
   script: |
     php composer.phar install --dev
-    php composer.phar global require phpstan/phpstan-shim:0.11.19
+    php composer.phar global require phpstan/phpstan-shim:0.12
     ln -s /builds/ps-addons/$CI_PROJECT_NAME /var/www/html/modules/$CI_PROJECT_NAME
-    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
+    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon .
```
Comment (Progi1984) [inline src/content/1.7/modules/testing/ci-cd.md:178]:
```suggestion
    php -d memory_limit=-1 ~/.composer/vendor/bin/phpstan analyse --configuration=/var/www/html/modules/$CI_PROJECT_NAME/tests/phpstan/phpstan.neon
```

No need with 0.12 version

**Review by PierreRambaud — APPROVED** [review]
I update the PR description and title :)

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @comxd

**Pull request merged**